client/my-sites/themes/trending-themes.jsx

```diff
@@ -0,0 +1,58 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { connect } from 'react-redux';
+/**
+ * Internal dependencies
+ */
+import { ConnectedThemesSelection } from './themes-selection';
+import { getTrendingThemes } from 'calypso/state/themes/actions';
+
+import {
+	getTrendingThemes as getTrendingThemesSelector,
+	areTrendingThemesLoading,
+} from 'calypso/state/themes/selectors';
+
+class TrendingThemes extends React.Component {
+	componentDidMount() {
+		if ( ! this.props.customizedThemesList.length ) {
+			this.fetchThemes();
+		}
+	}
+
+	componentDidUpdate( prevProps ) {
+		// Wait until rec themes to be loaded to scroll to search input if its in use.
+		const { isLoading, scrollToSearchInput, filter } = this.props;
+		if ( prevProps.isLoading !== isLoading && isLoading === false ) {
+			scrollToSearchInput();
+		}
+		if ( prevProps.filter !== filter ) {
+			this.fetchThemes();
+		}
+	}
+
+	fetchThemes() {
+		this.props.getTrendingThemes( this.props.filter );
+	}
+
+	render() {
+		return (
+			<>
```

Comment:
Nitpick: I don't believe this fragment is needed

Reply:
Good catch. I had c/p'd it from recommended-themes.jsx so I'll go ahead and remove it there too.